### PR TITLE
Create VPC

### PR DESCRIPTION
**Pull request by mnkap**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+resource "aws_subnet" "public1" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.1.0/24"
+
+  tags ={
+    Name = "public1"
+  }
+
+}
```
Comment (vladimirmukhin) [inline main.tf:27]:
```suggestion
  }
}
```

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+resource "aws_subnet" "private0" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.2.0/24"
+
+ tags ={
+    Name = "public0"
+  }
+
+}
```
Comment (vladimirmukhin) [inline main.tf:37]:
```suggestion
  }
}
```

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+resource "aws_subnet" "private1" {
+  vpc_id     = aws_vpc.main.id
+  cidr_block = "10.0.3.0/24"
+
+  tags ={
+    Name = "private1"
+  }
+
+}
```
Comment (vladimirmukhin) [inline main.tf:47]:
```suggestion
  }
}
```

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+resource "aws_internet_gateway" "main" {
+  vpc_id = aws_vpc.main.id
+
+
+  tags ={
+    Name = "main"
+  }
+
+}
```
Comment (vladimirmukhin) [inline main.tf:57]:
```suggestion
  }
}
```

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+resource "aws_eip" "nat1" {
+  vpc      = true
+
+  tags ={
+    Name = "nat1"
+  }
+  
+}
```
Comment (vladimirmukhin) [inline main.tf:74]:
```suggestion
  }
}
```

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+resource "aws_route_table_association" "private0" {
+  subnet_id      = aws_subnet.private0.id
+  route_table_id = aws_route_table.private0.id
+}
+
+resource "aws_route_table_association" "private1" {
+  subnet_id      = aws_subnet.private1.id
+  route_table_id = aws_route_table.private1.id
+}
```
Comment (vladimirmukhin) [inline main.tf:155]: Please end each file with a blank newline

File: provider.tf [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+  region = "us-east-1"
+}
```
Comment (vladimirmukhin) [inline provider.tf:13]: Please end each file with a blank newline

File: main.tf [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  route_table_id = aws_route_table.public.id
+}
+
+resource "aws_route_table_association" "public1" {
+  subnet_id      = aws_subnet.public1.id
+  route_table_id = aws_route_table.public.id
+}
+
+
+resource "aws_route_table_association" "private0" {
```
Comment (vladimirmukhin) [inline main.tf:147]:
```suggestion
}

resource "aws_route_table_association" "private0" {
```

File: main.tf [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+    gateway_id = aws_internet_gateway.main.id
+  }
+
+  tags ={
+    Name = "public"
+  }
+}
+
+
+resource "aws_route_table" "private0" {
```
Comment (vladimirmukhin) [inline main.tf:110]:
```suggestion
}

resource "aws_route_table" "private0" {
```

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+  allocation_id = aws_eip.nat1.id
+  subnet_id     = aws_subnet.public1.id
+
+  tags ={
+    Name = "main1"
+  }
+}
+
+
+resource "aws_route_table" "public" {
```
Comment (vladimirmukhin) [inline main.tf:96]:
```suggestion
}

resource "aws_route_table" "public" {
```

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
please run `terraform fmt` so terraform will automatically fix some code style mistakes and commit the result.

You can read more about terraform fmt here: https://www.terraform.io/cli/commands/fmt

**Review by vladimirmukhin — APPROVED** [review]

**Comment by mnkap** [comment]
Deploy VPC

**Comment by mnkap** [comment]
Deploy VPC

**Pull request merged**